src/main/java/Duke.java

```diff
@@ -1,10 +1,189 @@
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+
+    static void level2() {
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        List<String> lst = new ArrayList<>();
+
+        while (!input.equals("bye")) {
+            if (!input.equals("list")) {
+                lst.add(input);
+                System.out.println("added: " + input);
+            } else {
+                for (int i = 1; i <= lst.size(); i++) {
+                    System.out.println(i + ". " + lst.get(i-1));
+                }
+            }
+            input = sc.nextLine();
+        }
+
+        System.out.println("Bye. Hope to see you again soon!");
+
+    }
+
+
+    static void level3() {
+        List<Task> lst = new ArrayList<>();
+        lst.add(new Task("read book"));
+        lst.add(new Task("return book"));
+        lst.add(new Task("buy bread"));
+        lst.get(0).markAsDone();
+
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        String[] inputArray = input.split(" ");
+
+        while (true) {
+            if (input.equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+
+                for (int i = 1; i <= lst.size(); i++) {
+                    System.out.println(i + "." + lst.get(i - 1));
+                }
+            } else if (inputArray[0].equals("done")) {
+                System.out.println("Nice! I've marked this task as done:");
+                lst.get(Integer.parseInt(inputArray[1]) - 1).markAsDone();
+                System.out.println("  " + lst.get(Integer.parseInt(inputArray[1]) - 1));
+            }
+
+            input = sc.nextLine();
+            inputArray = input.split(" ");
+        }
+    }
+
+    static void level4() throws DukeException {
+        List<Task> lst = new ArrayList<>();
+        lst.add(new ToDoTask("read book"));
+        lst.add(new DeadlineTask("return book", "June 6th"));
+        lst.add(new EventTask("project meeting", "Aug 6th 2-4pm"));
+        lst.add(new ToDoTask("join sports club"));
+        lst.get(0).markAsDone();
+        lst.get(3).markAsDone();
+
+        Scanner sc = new Scanner(System.in);
+        String input;
+        String[] inputArray;
+
+        while (sc.hasNextLine()) {
+            input = sc.nextLine();
+            inputArray = input.split(" ");
+            if (inputArray[0].equals("list")) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 1; i <= lst.size(); i++) {
+                    System.out.println(i + "." + lst.get(i - 1));
+                }
+            } else if (inputArray[0].equals("todo") || inputArray[0].equals("deadline") || inputArray[0].equals("event")) { //adding a task
+                Task task = new Task("");
+                String[] inputArr1;
```

Comment:
perhaps extracting all the code that deals with different types of tasks could make your code clearer.